utils/generateHTML.js

```diff
@@ -1,58 +1,77 @@
 const { readFile, writeFile } = require("fs/promises");
 const path = require("path");
 const createHtml = require("create-html");
+var htmlBody = "";
 
 /**
  * Generates a new html file to a 'dist' directory.
  * @param {string} inputPath - A path to a text file
  * @param {string} stylesheetURL - A URL to a CSS stylesheet
  */
 async function generateHTMLFile(inputPath, stylesheetURL) {
-    try {
-        const filename = path.parse(inputPath).name;
-        const data = await readFile(inputPath, "utf-8");
+  try {
+    const filename = path.parse(inputPath).name;
+    const data = await readFile(inputPath, "utf-8");
 
-        // Retrieves HTML title (if any) - title is identified by being the first line of text followed by 2 blank lines.
-        const htmlTitle = data.match(/^.+(\r?\n\r?\n\r?\n)/);
-        // If there is an HTML title, wraps the title's content inside an <h1> tag.
-        const htmlBody =
-            htmlTitle == null
-                ? data
-                      .split(/\r?\n\r?\n/)
-                      .map((para) => {
-                          return `<p>${para.replace(/\r?\n/, " ")}</p>`;
-                      })
-                      .join(" ")
-                : `<h1>${htmlTitle[0].trim()}</h1> `.concat(
-                      data
-                          .substring(htmlTitle[0].length)
-                          .split(/\r?\n\r?\n/)
-                          .map((para) => {
-                              return `<p>${para.replace(/\r?\n/, " ")}</p>`;
-                          })
-                          .join(" ")
-                  );
+    // Retrieves HTML title (if any) - title is identified by being the first line of text followed by 2 blank lines.
+    const htmlTitle = data.match(/^.+(\r?\n\r?\n\r?\n)/);
+    // If there is an HTML title, wraps the title's content inside an <h1> tag.
 
-        const html = createHtml({
-            lang: "en",
-            head: `<meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    if (path.extname(inputPath) === ".txt") {
+      // handles ".txt" content
+      htmlBody =
+        htmlTitle == null
+          ? data
+              .split(/\r?\n\r?\n/)
+              .map((para) => {
+                return `<p>${para.replace(/\r?\n/, " ")}</p>\n\n`;
+              })
+              .join(" ")
+          : `<h1>${htmlTitle[0].trim()}</h1>\n\n`.concat(
+              data
+                .substring(htmlTitle[0].length)
+                .split(/\r?\n\r?\n/)
+                .map((para) => {
+                  return `<p>${para.replace(/\r?\n/, " ")}</p>\n\n`;
+                })
+                .join(" ")
+            );
+    } else {
+      //  handles ".md" content
+      htmlBody = data
+        .replace(/\[(.*?)\]\((.*?)\)/gim, "<a href='$2'>$1</a>") // replaces link -> <a href="$2">$1</a>
+        .split(/\r?\n\r?\n/)
+        .map(function (para) {
+          if (para.match(/(?<!#)#{1}\s/)) {
```

Comment:
From the Mozilla [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match):
> `String.match()` returns an Array whose contents depend on the presence or absence of the global (g) flag, or **null** if no matches are found. 

Thus, I suggest changing both the `if` and `else if` conditions to:
```js
if ([statement] != null) 
```